data/transactions/logic/eval.go

```diff
@@ -3372,11 +3579,28 @@ func opAppLocalPut(cx *EvalContext) {
 		return
 	}
 
-	addr, accountIdx, err := cx.accountReference(cx.stack[pprev])
-	if err == nil {
-		err = cx.Ledger.SetLocal(addr, key, sv.toTealValue(), accountIdx)
+	addr, accountIdx, err := cx.mutableAccountReference(cx.stack[pprev])
+	if err != nil {
+		cx.err = err
+		return
 	}
 
+	// if writing the same value, do nothing, matching ledger behavior with
+	// previous BuildEvalDelta mechanism
```

Comment:
I do not see where do we compare old and new in ledger:
SetKey:
```
	vdelta, ok := lsd.kvCow[key]
	if !ok {
		vdelta = valueDelta{old: oldValue, oldExists: oldOk}
	}
	vdelta.new = value
	vdelta.newExists = true
	lsd.kvCow[key] = vdelta
```
In `applyStorageDeltas` we write no matter what
```
states.KeyValue[k] = v.new
```

Reply:
appcow_test.go demonstrates that when a keys old and new value are the same, BuildEvalDelta does not spit out a stateDelta. (Lines 473, for example).

Maybe comes from valueDelta.serialize:
```
	if vd.oldExists && vd.old == vd.new {
		return
	}
```